test/tests/functional/pbs_hook_modifyvnode_state_changes.py

```diff
@@ -240,91 +236,100 @@ def test_hook_state_changes_00(self):
-        self.assertEqual(retpbsn['rc'], 0)
-        self.logger.debug("retpbsn=%s" % retpbsn)
+        nodeinfo = self.server.status(NODE)
+        self.logger.debug("nodeinfo=%s" % nodeinfo)
 
         # test effects of various state changes on each mom
-        for name, value in self.server.moms.items():
-            self.logger.debug("    ***%s:%s, type:%s" % (name, value,
-                              type(value)))
-            self.logger.debug("    ***%s:fqdn:    %s" % (name, value.fqdn))
-            self.logger.debug("    ***%s:hostname:%s" % (name, value.hostname))
-
+        for mom in self.server.moms.values():
             # State change test: mom stop
             start_time = int(time.time())
-            value.stop()
-            self.checkLog(start_time, value.fqdn, check_up=False,
+            mom.stop()
+            self.checkLog(start_time, mom.fqdn, check_up=False,
                           check_down=True)
             self.server.log_match("v.state_hex=0x2 v_o.state_hex=0x0",
                                   starttime=start_time)
 
             # State change test: mom start
             start_time = int(time.time())
-            value.start()
-            self.checkLog(start_time, value.fqdn, check_up=True,
+            mom.start()
+            self.checkLog(start_time, mom.fqdn, check_up=True,
                           check_down=False)
             self.server.log_match("v.state_hex=0x0 v_o.state_hex=0x400",
                                   starttime=start_time)
 
             # State change test: mom restart
             start_time = int(time.time())
-            value.restart()
-            self.checkLog(start_time, value.fqdn, check_up=True,
+            mom.restart()
+            self.checkLog(start_time, mom.fqdn, check_up=True,
                           check_down=True)
             self.server.log_match("v.state_hex=0x2 v_o.state_hex=0x0",
                                   starttime=start_time)
             self.server.log_match("v.state_hex=0x0 v_o.state_hex=0x400",
                                   starttime=start_time)
 
-            # State change test: take mom offline (remove mom)
+            # State change test: take mom offline and online using pbsnodes
+            
+            # take offline
             start_time = int(time.time())
-            self.logger.debug("    ***offline mom:%s" % value)
+            self.logger.debug("    ***offline mom:%s" % mom)
             pbsnodesoffline = os.path.join(self.server.pbs_conf['PBS_EXEC'],
                                            'bin', 'pbsnodes -o %s' %
-                                           value.fqdn)
+                                           mom.shortname)
             self.logger.debug("pbsnodesoffline=%s" % pbsnodesoffline)
-            retpbsn = self.du.run_cmd(self.server.hostname, pbsnodesoffline,
-                                      sudo=True)
+            retpbsn = self.du.run_cmd(self.server.hostname, pbsnodesoffline)
```

Comment:
I will remove the pbsnodes tests if preferred, but am curious to understand the reason. It would seem pbsnodes follows a different execution path (at least client-wise) from the self.server.manager() calls I added, so why not test both pathways?